src/validateChangelog.test.js

```diff
@@ -0,0 +1,445 @@
+      }),
+    ).toThrow('Failed to find Unreleased header');
+  });
+
+  it('should throw if the wrong repo URL is used', () => {
+    expect(() =>
+      validateChangelog({
+        changelogContent: changelogWithReleases,
+        currentVersion: '1.0.0',
+        repoUrl: 'https://github.com/DifferentOrganization/DifferentRepository',
+        isReleaseCandidate: false,
+      }),
+    ).toThrow('Changelog is not well-formatted');
+  });
+
+  it('should throw if a comparison release link is missing', () => {
+    const changelogWithoutReleaseLink = changelogWithReleases.replace(
+      '[1.0.0]: https://github.com/ExampleUsernameOrOrganization/ExampleRepository/compare/v0.0.2...v1.0.0\n',
+      '',
+    );
+    expect(() =>
+      validateChangelog({
+        changelogContent: changelogWithoutReleaseLink,
+        currentVersion: '1.0.0',
+        repoUrl:
+          'https://github.com/ExampleUsernameOrOrganization/ExampleRepository',
+        isReleaseCandidate: false,
+      }),
+    ).toThrow('Changelog is not well-formatted');
+  });
+
+  it('should throw if the first release link is missing', () => {
+    const changelogWithoutFirstReleaseLink = changelogWithReleases.replace(
+      '[0.0.1]: https://github.com/ExampleUsernameOrOrganization/ExampleRepository/releases/tag/v0.0.1\n',
+      '',
+    );
+    expect(() =>
+      validateChangelog({
+        changelogContent: changelogWithoutFirstReleaseLink,
+        currentVersion: '1.0.0',
+        repoUrl:
+          'https://github.com/ExampleUsernameOrOrganization/ExampleRepository',
+        isReleaseCandidate: false,
+      }),
+    ).toThrow('Changelog is not well-formatted');
+  });
+
+  it('should throw if release links are in a different order than the release headers', () => {
+    const thirdReleaseLink =
+      '[1.0.0]: https://github.com/ExampleUsernameOrOrganization/ExampleRepository/compare/v0.0.2...v1.0.0';
+    const secondReleaseLink =
+      '[0.0.2]: https://github.com/ExampleUsernameOrOrganization/ExampleRepository/compare/v0.0.1...v0.0.2';
+    const changelogWithoutFirstReleaseLink = changelogWithReleases.replace(
+      `${thirdReleaseLink}\n${secondReleaseLink}`,
+      `${secondReleaseLink}\n${thirdReleaseLink}`,
+    );
+    expect(() =>
+      validateChangelog({
+        changelogContent: changelogWithoutFirstReleaseLink,
+        currentVersion: '1.0.0',
+        repoUrl:
+          'https://github.com/ExampleUsernameOrOrganization/ExampleRepository',
+        isReleaseCandidate: false,
+      }),
+    ).toThrow('Changelog is not well-formatted');
+  });
+
+  it('should not throw for changelog with branching releases', () => {
+    expect(() =>
+      validateChangelog({
+        changelogContent: branchingChangelog,
+        currentVersion: '1.0.0',
+        repoUrl:
+          'https://github.com/ExampleUsernameOrOrganization/ExampleRepository',
+        isReleaseCandidate: false,
+      }),
+    ).not.toThrow();
+  });
```

Comment:
nit: I was surprised to find a success / `not.toThrow()` case here. It might be better to group them.

Reply:
I had thought these were ordered by feature, roughly? But looking at it now, that's not clear either. I'll try to order them a bit more sensibly.

Reply:
Tests re-ordered in https://github.com/MetaMask/auto-changelog/pull/28/commits/bcebb91c2ddcb6e7fcaf321472f8526a7dd8e502